include/boost/url/experimental/detail/router.hpp

```diff
@@ -0,0 +1,1230 @@
+                    *make_pct_string_view_unsafe(
+                        expi.base(),
+                        dseg.end().base() - expi.base(),
+                        dseg.size() - dp);
+                if (exp.starts_with("+") || exp.starts_with("-"))
+                    exp.remove_prefix(1);
+                if (!std::all_of(
+                        exp.begin(), exp.end(),
+                        grammar::lut_chars("0123456789")))
+                    return false;
+                --expi;
+                --dp;
+            }
+            dseg = *make_pct_string_view_unsafe(
+                    dseg.begin().base(),
+                    expi.base() - dseg.begin().base(),
+                    dp);
+            return true;
+        };
+
+        switch (type_)
+        {
+        case 'a':
+            remove_exp_suffix(dseg, 'p');
+            break;
+        case 'A':
+            remove_exp_suffix(dseg, 'P');
+            break;
+        case 'e':
+        case 'g':
+            remove_exp_suffix(dseg, 'e');
+            break;
+        case 'E':
+        case 'G':
+            remove_exp_suffix(dseg, 'E');
+            break;
+        }
+
+        if (precision_)
+        {
+            if (*precision_ == 0)
+            {
+                if (dseg.find('.') != dseg.end())
+                {
+                    return false;
+                }
+            }
+            else
+            {
+                auto p = dseg.find('.');
+                if (p == dseg.end())
+                    return false;
+                std::size_t d = 0;
+                ++p;
+                while (p != dseg.end())
+                {
+                    ++d;
+                    ++p;
+                }
+                if (d != *precision_)
+                    return false;
+            }
+        }
+
+        auto is_token =[](
+            decode_view dseg,
+            grammar::lut_chars const& cs)
+        {
+            return std::all_of(
+                dseg.begin(), dseg.end(), cs);
+        };
+
+        auto is_float_token =[](
+            decode_view dseg,
+            grammar::lut_chars const& cs,
+            char point,
+            char alternate)
+        {
+            auto it = dseg.begin();
+            auto end = dseg.end();
+            it = std::find_if_not(it, end, cs);
+
+            // alternate form for floating-point
+            // numbers requires decimal point
+            if (alternate &&
+                it == end)
+                return false;
+
+            if (
+                it != end &&
+                *it == point)
+            {
+                ++it;
+                it = std::find_if_not(it, end, cs);
+            }
+            return it == end;
+        };
+
+        char point = '.';
+        if (use_locale_)
+        {
+            point = std::use_facet<
+                        std::numpunct<char>>(std::locale()).decimal_point();
+        }
+
+        switch (type_)
```

Comment:
We are only supporting strings and unsigned... all the rest of this has to be removed. I guess you could save it to your personal github account as a souvenir.